server/tracing/tracing.go

```diff
@@ -0,0 +1,52 @@
+package tracing
+
+import (
+	"github.com/tigrisdata/tigris/server/config"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/profiler"
+)
+
+func getTracingOptions(config *config.Config) []tracer.StartOption {
+	var opts []tracer.StartOption
+	opts = append(opts, tracer.WithTraceEnabled(config.Tracing.Enabled))
+	opts = append(opts, tracer.WithService(config.Tags.Service))
+	opts = append(opts, tracer.WithEnv(config.Tags.Environment))
+	opts = append(opts, tracer.WithServiceVersion(config.Tags.Version))
+	if config.Tracing.WithUDS != "" {
+		opts = append(opts, tracer.WithUDS(config.Tracing.WithUDS))
+	}
+	if config.Tracing.WithAgentAddr != "" {
+		opts = append(opts, tracer.WithAgentAddr(config.Tracing.WithAgentAddr))
+	}
+	if config.Tracing.WithDogStatsdAddr != "" {
+		opts = append(opts, tracer.WithAgentAddr(config.Tracing.WithAgentAddr))
+	}
+
+	return opts
+}
+
+func getProfilingOptions(config *config.Config) []profiler.Option {
+	var opts []profiler.Option
+	opts = append(opts, profiler.WithService(config.Tags.Service))
+	opts = append(opts, profiler.WithEnv(config.Tags.Environment))
+	opts = append(opts, profiler.WithVersion(config.Tags.Version))
+	return opts
+}
+
+func InitTracer(config *config.Config) error {
+	if !config.Profiling.Enabled {
+		return nil
+	}
+
+	tracer.Start(getTracingOptions(config)...)
+	defer tracer.Stop()
+
+	if config.Profiling.Enabled {
```

Comment:
you don't need this `if` it has been take care of on line 37

Reply:
You actually caught a 🐞 here. L37 should say `config.Tracing.Enabled`! Great catch!

Reply:
It looks correct to me. if not enabled return.